File: Year 3/React Native/Simple SMHI Weather Warning application/my-second-project/models/WarningAreas.ts
import { Area } from "./Area";
import { AreaName } from "./AreaName";
import { EventDescription } from "./EventDescription";
import { WarningLevel } from "./WarningLevel";

export class WarningAreas {
    id: Number;
    area?: Area
    areaName: AreaName;
    approximateStart: string;
    approximateEnd: string;
    normalProbability: boolean;
    warningLevel: WarningLevel;
    eventDescription: EventDescription;
    published: string;
    descriptions: any[];
    affectedAreas: any[];
    constructor(
        id:number = 0,
        area: Area = {} as Area,
        areaName?: AreaName,
        approximateStart: string = '',
        approximateEnd: string = '',
        normalProbability: boolean = false,
        warningLevel?: WarningLevel,
        eventDescription?: EventDescription,
        published: string = '',
        descriptions: any[] = [],
        affectedAreas: any[] = []
    ) {
        this.id = id;
        this.area = area;
        this.areaName = new AreaName;
        this.approximateStart = approximateStart;
        this.approximateEnd = approximateEnd;
        this.normalProbability = normalProbability;
        this.warningLevel = new WarningLevel;
        this.eventDescription = new EventDescription;
        this.published = published;
        this.descriptions = descriptions;
        this.affectedAreas = affectedAreas;
    }
    getId() {
        return this.id;
    }

    setId(value: number) {
        this.id = value;
    }

    getArea() {
        return this.area;
    }

    setArea(value: Area | undefined) {
        this.area = value;
    }

    getAreaName() {
        return this.areaName;
    }

    setAreaName(value: AreaName) {
        this.areaName = value;
    }

    getApproximateStart() {
        return this.approximateStart;
    }

    setApproximateStart(value: string) {
        this.approximateStart = value;
    }

    getApproximateEnd() {
        return this.approximateEnd;
    }

    setApproximateEnd(value: string) {
        this.approximateEnd = value;
    }

    getNormalProbability() {
        return this.normalProbability;
    }

    setNormalProbability(value: boolean) {
        this.normalProbability = value;
    }

    getWarningLevel() {
        return this.warningLevel;
    }

    setWarningLevel(value: WarningLevel) {
        this.warningLevel = value;
    }

    getEventDescription() {
        return this.eventDescription;
    }

    setEventDescription(value: EventDescription) {
        this.eventDescription = value;
    }

    getPublished() {
        return this.published;
    }

    setPublished(value: string) {
        this.published = value;
    }

    getDescriptions() {
        return this.descriptions;
    }

    setDescriptions(value: any[]) {
        this.descriptions = value;
    }

    getAffectedAreas() {
        return this.affectedAreas;
    }

    setAffectedAreas(value: any[]) {
        this.affectedAreas = value;
    }
}

File: Year 3/Android Java/Simple SMHI Warning App JavaVersion/GeoWarning/app/src/main/java/com/example/geowarning/model/WarningAreas.java
package com.example.geowarning.model;

import java.util.List;

public class WarningAreas {
    private int id; // Unique ID for each alert
    private Area area; // GeoJSON object (you may need to create a separate class for this)
    private AreaName areaName; // AreaName object (if applicable)
    private String approximateStart; // Date string for when the warning area takes effect
    private String approximateEnd; // Date string for when the warning area expires (optional)
    private boolean normalProbability; // Describes whether there is a normal probability
    private WarningLevel warningLevel; // Warning level object
    private EventDescription eventDescription; // EventDescription object
    private String published; // Date string for when the warning area was published
    private List<Description> descriptions; // List of descriptions (if applicable)
    private List<AffectedArea> affectedAreas; // List of affected areas

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Area getArea() {
        return area;
    }

    public void setArea(Area area) {
        this.area = area;
    }

    public AreaName getAreaName() {
        return areaName;
    }

    public void setAreaName(AreaName areaName) {
        this.areaName = areaName;
    }

    public String getApproximateStart() {
        return approximateStart;
    }

    public void setApproximateStart(String approximateStart) {
        this.approximateStart = approximateStart;
    }

    public String getApproximateEnd() {
        return approximateEnd;
    }

    public void setApproximateEnd(String approximateEnd) {
        this.approximateEnd = approximateEnd;
    }

    public boolean isNormalProbability() {
        return normalProbability;
    }

    public void setNormalProbability(boolean normalProbability) {
        this.normalProbability = normalProbability;
    }

    public WarningLevel getWarningLevel() {
        return warningLevel;
    }

    public void setWarningLevel(WarningLevel warningLevel) {
        this.warningLevel = warningLevel;
    }

    public EventDescription getEventDescription() {
        return eventDescription;
    }

    public void setEventDescription(EventDescription eventDescription) {
        this.eventDescription = eventDescription;
    }

    public String getPublished() {
        return published;
    }

    public void setPublished(String published) {
        this.published = published;
    }

    public List<Description> getDescriptions() {
        return descriptions;
    }

    public void setDescriptions(List<Description> descriptions) {
        this.descriptions = descriptions;
    }

    public List<AffectedArea> getAffectedAreas() {
        return affectedAreas;
    }

    public void setAffectedAreas(List<AffectedArea> affectedAreas) {
        this.affectedAreas = affectedAreas;
    }
}
